### PR TITLE
minor improvements

### PR DESCRIPTION
**Pull request by jerger**
* use only 8 char format for importing keys  (found the full blown fingerprint not working for "gpg (GnuPG) 2.1.11" on ubuntu 16.04)
* added publish & reencrypt - usefull for work in our team

hkp://*:80 is needed due to our company firewall ..

### CODE REVIEW

**Comment by TheRealBro** [comment]
> use only 8 char format for importing keys (found the full blown fingerprint not working for "gpg (GnuPG) 2.1.11" on ubuntu 16.04)
👎 ❗️ 

@jerger this is a very bad Idea - as there have been found hash collisions for the "short key" format of gpg... if gpg-keys are searched or used with the ID - always the long-key format should be used - especially like in this case where security in password db is somewhat important.

**Comment by TheRealBro** [comment]
> hkp://*:80 is needed due to our company firewall ..

yay ... your firewall ... don't do that in upstream, fix the firewall or fix it in a cloned/forked repo on your site

**Comment by jerger** [comment]
On your comment about upstream I agree :-)

In general I agree on your comment about hashes.
In our specific situation we've to find a solution with gpg2 issues (yes we know the weakness and yes, we do close reviews on this):
```
gpg2 --keyserver hkp://keyserver.ubuntu.com --search-keys 0x8....D
gpg: error searching keyserver: No data
gpg: keyserver search failed: No data
```
Short ID's are working, do you have in idea about?

**Comment by TheRealBro** [comment]
@jerger 
sorry for the late replay, fine that you changed the critical points regarding port 80.
I still find it not a very neat solution to do a "cut" on the .gpg-id because, it should "just work" with every format provided here.

My tests with the following versions

```
gpg (GnuPG) 2.2.5
```
on openSUSE Tumbleweed & Leap 15

and I tested it with openSUSE Leap 42.3 with the following version

```
gpg (GnuPG) 2.0.24
```

All of the three commands below worked (and brought back my GPG key as expected):

```
#Short ID format
gpg2 --keyserver hkp://keyserver.opensuse.org --search-keys 0xC930E5E6

#Long ID format
gpg2 --keyserver hkp://keyserver.opensuse.org --search-keys 0x433A21C2C930E5E6

#Full fingerprint format
gpg2 --keyserver hkp://keyserver.opensuse.org --search-keys 0x1D09FFA7B54FC570309202D4433A21C2C930E5E6
```

File: Makefile [hunk 1]
```diff
@@ -12,7 +12,7 @@ help:
 
 .PHONY: import-and-sign
 import-and-sign: ## Import in GPG all keys from the list of allowed keys
-	$(foreach var,$(shell find . -name .gpg-id | xargs cat | sort | uniq), \
+	$(foreach var,$(shell find . -name .gpg-id | xargs cat | sort | uniq | cut -c 33-40), \
```
Comment (TheRealBro) [inline Makefile:15]: this cut here is very dangerous regarding "breaking things" and regarding security purpose of this tool

**Comment by jerger** [comment]
No Problem :-)
I assume, that the problem is located on server-side (but did not really track this down). 
We've observed, that sometimes the query on  hkp://keyserver.ubuntu.com works sometimes not (our team is distributed, so not everybody is directed to the same ip). In meantime, we've switched to gopass. gopass distributes known keys through a versioned .gpg-keys directory ... I think I like this solution even more than my PR :-)

**Review by TheRealBro — CHANGES_REQUESTED** [review]
You maybe want to close this PR than @jerger - or change it as the reencrypt and publish-public-keys functions might be handy for some users :).

**Pull request closed without merging**